src/main/java/org/sagebionetworks/web/client/presenter/SearchPresenter.java

```diff
@@ -414,4 +449,141 @@ private static String join(List<String> list, String delimiter) {
   private String createTimeValueKey(String facetName, String facetValue) {
     return facetName + facetValue;
   }
+
+  private long getLimit() {
+    return currentSearch.getSize() == null ? 10L : currentSearch.getSize();
+  }
+
+  private int getIndexOfHit(Hit targetHit) {
+    int cumulativeIndex = 0; // Note: 0-indexed
+    for (SearchResults results : allPagesOfResults) {
+      List<Hit> hits = results.getHits();
+      int indexInPage = hits.indexOf(targetHit);
+      if (indexInPage != -1) {
+        return cumulativeIndex + indexInPage;
+      }
+      cumulativeIndex += hits.size();
+    }
+    return -1; // Hit not found
+  }
+
+  private int getIndexOfPageContainingHit(Hit targetHit) {
+    // Note: 0-indexed
+    for (int pageIndex = 0; pageIndex < allPagesOfResults.size(); pageIndex++) {
+      List<Hit> hits = allPagesOfResults.get(pageIndex).getHits();
+      if (hits.contains(targetHit)) {
+        return pageIndex;
+      }
+    }
+    return -1; // Hit not found
+  }
+
+  /**
+   * Get event data for an individual search result
+   * @param hit
+   * @return
+   */
+  private SearchResultEventData getSearchResultEventData(Hit hit) {
+    SearchResultEventData resultEventData =
+      addSearchPageResultToAnalyticsEventData(new SearchResultEventData());
+    resultEventData.page_index =
+      Double.valueOf(1 + getIndexOfPageContainingHit(hit));
+    resultEventData.rank = Double.valueOf(1 + getIndexOfHit(hit));
+    resultEventData.item_type = SearchItemType.entity.toString();
+    resultEventData.item_id = hit.getId();
+    return resultEventData;
+  }
+
+  @Override
+  public ClickHandler getSearchResultClickedHandler(Hit hit) {
+    return event -> {
+      SearchResultEventData resultEventData = getSearchResultEventData(hit);
+      searchAnalyticsClient.sendSearchResultClickedEvent(resultEventData);
+    };
+  }
+
+  /**
+   * Adds data related to the latest search request that is not dependent on page
+   * @param eventData
+   */
+  private <T extends SearchQueryEventData> T addSearchToAnalyticsEventData(
+    T eventData
+  ) {
+    eventData.query_term = join(currentSearch.getQueryTerm(), " ");
```

Comment:
Would this be more informative if we used a different separator to join the query terms?

Reply:
Probably so, as we could capture cases where uses use things like quotation marks, or perhaps future changes are made so we do not split on the space token. We could use some other separator but we would have to make sure to escape it when we serialize it. It's a bit frustrating that GA4 only allows string values for custom events.

Reply:
It will probably depend on the viewer.  
But stringifying the JSON array seems like the best thing we can do to ensure they have all the necessary info.